### PR TITLE
feat(proxy): persist payload cleanup savings metrics in usage snapshots (#103)

### LINKED ISSUE #103: Post-alpha: add payload cleanup savings metrics to usage snapshots
## Context\nRoadmap source: PennyPrompt-v2.md (Semanas 7-12, semana 7 payload cleanup).\n\n## Goal\nMake payload cleanup impact measurable by recording byte-level savings metadata during proxy response processing.\n\n## Scope\n- Track bytes before/after cleanup for non-stream and streaming responses.\n- Attach cleanup metrics into persisted request usage pricing snapshot metadata.\n- Keep API response compatibility unchanged (no breaking schema changes in proxy output).\n- Add tests for metric recording and zero-savings behavior.\n\n## Acceptance Criteria\n- Usage snapshots persist cleanup metadata when cleanup is enabled.\n- Metrics include bytes_in, bytes_out, and bytes_saved (or equivalent).\n- Stream path accumulates savings across chunks.\n- Existing CI checks stay green.

### PR DESCRIPTION
**Pull request by manuelpenazuniga**
## Summary
- Add payload cleanup savings metrics persistence to usage pricing snapshots.
- Record cleanup metadata under `payload_cleanup`:
  - `enabled`
  - `strip_ansi`
  - `minify_json`
  - `bytes_in`
  - `bytes_out`
  - `bytes_saved`
- Support both non-stream and streaming paths.
- Accumulate stream cleanup byte metrics across SSE chunks.
- Add tests for metrics persistence in non-stream and stream scenarios.

## Validation
- cargo fmt --all
- cargo test -p penny-proxy
- cargo check --workspace --locked

Closes #103

### CODE REVIEW

File: crates/penny-proxy/src/lib.rs [hunk 1]
```diff
@@ -648,38 +679,83 @@ fn minify_json_string(input: &str) -> Option<String> {
...
-        return if cleanup.strip_ansi {
+        let cleaned = if cleanup.strip_ansi {
             strip_ansi_sequences(line)
         } else {
             line.to_string()
         };
+        return (
+            cleaned.clone(),
+            CleanupMetrics::from_lengths(line.len(), cleaned.len()),
+        );
```
[the remark this answers is not in the document]
Comment (manuelpenazuniga) [reply crates/penny-proxy/src/lib.rs:698]: ok

File: crates/penny-proxy/src/lib.rs [hunk 2]
```diff
@@ -648,38 +679,83 @@ fn minify_json_string(input: &str) -> Option<String> {
...
```
[the remark this answers is not in the document]
Comment (manuelpenazuniga) [reply crates/penny-proxy/src/lib.rs:713]: ok

File: crates/penny-proxy/src/lib.rs [hunk 3]
```diff
@@ -648,38 +679,83 @@ fn minify_json_string(input: &str) -> Option<String> {
...
     apply_cleanup_to_json(&mut parsed, cleanup);
     match serde_json::to_string(&parsed) {
-        Ok(serialized) => format!("data: {serialized}{trailing_newline}"),
-        Err(_) => line.to_string(),
+        Ok(serialized) => {
+            let cleaned = format!("data: {serialized}{trailing_newline}");
+            (
+                cleaned.clone(),
+                CleanupMetrics::from_lengths(line.len(), cleaned.len()),
+            )
```
[the remark this answers is not in the document]
Comment (manuelpenazuniga) [reply crates/penny-proxy/src/lib.rs:722]: okok
